docs/DeveloperGuide.md

```diff
@@ -6,7 +6,152 @@
 
 ## Design & implementation
```

Comment:
Might want to highlight value proposition as it is graded
![image](https://user-images.githubusercontent.com/65325165/160959907-e69a7b07-867e-4eb3-839b-54aaee3c2c3c.png)